server/apiserver/argoserver.go

```diff
@@ -124,9 +124,9 @@ func init() {
 	}
 }
 
-func getResourceCacheNamespace(opts ArgoServerOpts) string {
-	if opts.ManagedNamespace != "" {
-		return opts.ManagedNamespace
+func getResourceCacheNamespace(managedNamespace string) string {
+	if managedNamespace != "" {
+		return managedNamespace
```

Comment:
this logic is not quite correct (potentially due to #9989?) -- if not `managedNamespace`, it should check if `namespaced`, and, if so, use `namespace`, otherwise use cluster scope. Same as I wrote in https://github.com/argoproj/argo-workflows/issues/13177#issuecomment-2167280162.

Right now this is missing namespaced mode without managed namespace

Reply:
Nevermind, apparently the CLI init handles this with [`managedNamespace = namespace`](https://github.com/argoproj/argo-workflows/blob/0ca0c0f721c9bf05c63dea98168cfa87441656c7/cmd/argo/commands/server.go#L98) in that case.

Reply:
Thank you for the review, yes, CLI init handles that case.